### PR TITLE
Added integration tests for Memcached. Fixed NotNull issue

### PR DESCRIPTION
**Pull request by mmoayyed**
Handles https://github.com/Jasig/cas/issues/942

### CODE REVIEW

**Comment by leleuj** [comment]
+1

File: cas-server-integration-memcached/pom.xml [hunk 1]
```diff
@@ -52,35 +52,44 @@
...
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
-      <version>1.9.0</version>
-      <scope>test</scope>
-    </dependency>
+        <!-- Test dependencies -->
+        <dependency>
+            <groupId>de.flapdoodle.embed</groupId>
+            <artifactId>de.flapdoodle.embed.memcached</artifactId>
```
Comment (serac) [inline cas-server-integration-memcached/pom.xml]: An embedded memcached! Nice.

**Comment by serac** [comment]
+1. I think we should take this opportunity to fix the whitespace in the pom. Or we could do a global pom whitespace cleanup prior to 4.1 release. I recall lots of pom.xml files have mixed tabs/spaces, which we really ought to clean up.

**Comment by mmoayyed** [comment]
:+1: Fixing whitespaces would be great! I'll create an issue. Coincidentally, I realized most if not all poms have duplicate group ids, which should be cleaned as well.

**Pull request merged**